docs/source/traits_user_manual/advanced.rst

```diff
@@ -301,6 +301,38 @@ exception, as does attempting to set an attribute that is not one of the three
 defined attributes. In essence, TreeNode behaves like a type-checked data
 structure.
 
+.. index:: HasRequiredTraits class
+
+.. _hasrequiredtraits:
+
+HasRequiredTraits
+'''''''''''''''''
+
+This class builds on the functionality of HasStrictTraits and ensures
```

Comment:
It might make sense to replace the somewhat vague "builds on" with something more precise. For example "This class inherits from ``HasStrictTraits`` [...]". Or even: "This subclass of ``HasStrictTraits`` ensures that ...".